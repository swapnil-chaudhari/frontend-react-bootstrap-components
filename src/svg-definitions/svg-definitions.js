import React from 'react';
import {
    SVGAdvertise,
    SVGArtworkEmpty,
    SVGAttention,
    SVGAudioDisabled,
    SVGAudioEnabled,
    SVGAudioPlayerMute,
    SVGAudioPlayerNext,
    SVGAudioPlayerNextLite,
    SVGAudioPlayerPause,
    SVGAudioPlayerPauseLite,
    SVGAudioPlayerPlay,
    SVGAudioPlayerPrevious,
    SVGAudioPlayerPreviousLite,
    SVGAudioPlayerVolume,
    SVGCalendar,
    SVGChevron,
    SVGCleanVersion,
    SVGCloseButton,
    SVGCloseRound,
    SVGCompleteCheckmark,
    SVGDigital,
    SVGDigitalAudio,
    SVGDownDirectionArrow,
    SVGDownloadBox,
    SVGDownloadSymbol,
    SVGEdit,
    SVGError404,
    SVGExplicit,
    SVGExternalLink,
    SVGFilledChevronDown,
    SVGFilledChevronUp,
    SVGFilm,
    SVGGear,
    SVGGoToCatalog,
    SVGHistory,
    SVGInfo,
    SVGLeftPaginationArrow,
    SVGLoadingArrows,
    SVGLoadingRing,
    SVGLogo,
    SVGLogoAndName,
    SVGMan,
    SVGNewProduct,
    SVGPdf,
    SVGPhysical,
    SVGPhysicalAudio,
    SVGPlay,
    SVGPlus,
    SVGRefresh,
    SVGRightDirectionArrow,
    SVGRightPaginationArrow,
    SVGRoyaltyShareLogo,
    SVGSearch,
    SVGServerError,
    SVGSpotify,
    SVGStatusComplete,
    SVGStatusInProgress,
    SVGStatusRed,
    SVGStatusSubmitted,
    SVGStop,
    SVGSuccessCheckmark,
    SVGThreeDots,
    SVGThreeRightDirectedArrows,
    SVGTracksCompleteCheckmark,
    SVGTracksEmpty,
    SVGTransparentLogo,
    SVGTrashCan,
    SVGTripleCheckmark,
    SVGUpAndDownArrows,
    SVGUpAndDownCarets,
    SVGUpDirectionArrow,
    SVGUpload,
    SVGViewMoreNextArrow,
    SVGWoman,
    SVGXls,
    SVGYoutube,
    SVGTwitter,
    SVGAan,
    SVGFacebook,
    SVGCamera,
    SVGPlaylistPlaceholder,
    SVGUploadArrow,
    SVGImportArrow,
    SVGVideo,
    SVGVideoError,
} from './svgs';

const SVGDefinitions = () =>
    <svg style={ { display: 'block', width: 0, height: 0 } } xmlns="http://www.w3.org/2000/svg">
        <SVGAan />
        <SVGAdvertise />
        <SVGArtworkEmpty />
        <SVGAttention />
        <SVGAudioDisabled />
        <SVGAudioEnabled />
        <SVGAudioPlayerMute />
        <SVGAudioPlayerNext />
        <SVGAudioPlayerNextLite />
        <SVGAudioPlayerPause />
        <SVGAudioPlayerPauseLite />
        <SVGAudioPlayerPlay />
        <SVGAudioPlayerPrevious />
        <SVGAudioPlayerPreviousLite />
        <SVGAudioPlayerVolume />
        <SVGCalendar />
        <SVGChevron />
        <SVGCleanVersion />
        <SVGCloseButton />
        <SVGCloseRound />
        <SVGCompleteCheckmark />
        <SVGDigital />
        <SVGDigitalAudio />
        <SVGDownDirectionArrow />
        <SVGDownloadBox />
        <SVGDownloadSymbol />
        <SVGEdit />
        <SVGError404 />
        <SVGExplicit />
        <SVGExternalLink />
        <SVGFacebook />
        <SVGFilledChevronDown />
        <SVGFilledChevronUp />
        <SVGFilm />
        <SVGGear />
        <SVGGoToCatalog />
        <SVGHistory />
        <SVGInfo />
        <SVGLeftPaginationArrow />
        <SVGLoadingArrows />
        <SVGLoadingRing />
        <SVGLogo />
        <SVGLogoAndName />
        <SVGMan />
        <SVGNewProduct />
        <SVGPdf />
        <SVGPhysical />
        <SVGPhysicalAudio />
        <SVGPlay />
        <SVGPlus />
        <SVGRefresh />
        <SVGRightDirectionArrow />
        <SVGRightPaginationArrow />
        <SVGRoyaltyShareLogo />
        <SVGSearch />
        <SVGServerError />
        <SVGSpotify />
        <SVGStatusComplete />
        <SVGStatusInProgress />
        <SVGStatusRed />
        <SVGStatusSubmitted />
        <SVGStop />
        <SVGSuccessCheckmark />
        <SVGThreeDots />
        <SVGThreeRightDirectedArrows />
        <SVGTracksCompleteCheckmark />
        <SVGTracksEmpty />
        <SVGTransparentLogo />
        <SVGTrashCan />
        <SVGTwitter />
        <SVGTripleCheckmark />
        <SVGUpAndDownArrows />
        <SVGUpAndDownCarets />
        <SVGUpDirectionArrow />
        <SVGUpload />
        <SVGViewMoreNextArrow />
        <SVGWoman />
        <SVGXls />
        <SVGYoutube />
        <SVGCamera />
        <SVGPlaylistPlaceholder />
        <SVGUploadArrow />
        <SVGImportArrow />
        <SVGVideo />
        <SVGVideoError />
    </svg>;

export default SVGDefinitions;
