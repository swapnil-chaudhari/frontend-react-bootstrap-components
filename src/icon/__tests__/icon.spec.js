import React from 'react';
import Icon from '../icon';
import renderShallow from 'render-shallow';
import { ICON_IDS } from 'src/svg-definitions/svgs';

describe('<Icon>', () => {

    describe('when it renders', () => {
        const id = ICON_IDS.CALENDAR;
        let component;

        beforeAll(() => {
            component = renderShallow(<Icon id={ id } />).output;
        });

        test('renders an <svg> with the id from props.id', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('when it renders with a className prop', () => {
        const className = 'some-more-specific-class';
        const id = ICON_IDS.CALENDAR;
        let component;

        beforeAll(() => {
            component = renderShallow(<Icon className={ className } id={ id } />).output;
        });

        test('renders an <svg> with the additional className(s)', () => {
            expect(component).toMatchSnapshot();
        });
    });

    describe('when it renders with an onClick prop', () => {
        const props = {
            id: ICON_IDS.CALENDAR,
            onClick: () => {}
        };
        let component;

        beforeAll(() => {
            component = renderShallow(<Icon { ...props } />).output;
        });

        test('renders an <Icon> and passes the onClick prop to it', () => {
            expect(component.props.onClick).toEqual(props.onClick);
        });
    });

});
