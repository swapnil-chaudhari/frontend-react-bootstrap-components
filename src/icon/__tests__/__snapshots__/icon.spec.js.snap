// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon> when it renders renders an <svg> with the id from props.id 1`] = `
<svg
  className=""
  height={undefined}
  onClick={undefined}
  width={undefined}
  x={undefined}
  y={undefined}
>
  <use
    xlinkHref="#icon-calendar"
  />
</svg>
`;

exports[`<Icon> when it renders with a className prop renders an <svg> with the additional className(s) 1`] = `
<svg
  className="some-more-specific-class"
  height={undefined}
  onClick={undefined}
  width={undefined}
  x={undefined}
  y={undefined}
>
  <use
    xlinkHref="#icon-calendar"
  />
</svg>
`;
